amanuensis/resources/admin/admin_associated_user.py

```diff
@@ -82,12 +82,10 @@ def add_associated_users(users, role=None):
             
             #get the foreign key for the role
             if not role:
-                role_id = session.query(AssociatedUserRoles.id).filter(AssociatedUserRoles.code == "METADATA_ACCESS").first()[0]
+                role_id = udm.get_associated_user_role_by_code(session).id
```

Comment:
Isn't this supposed to have a code as parameter if it is by code?

Reply:
Ah I see you defaulted it later in the udm function. I would try to keep the function in the UDM as general as possible and would add the default role in the config. Read from the config here and pass it as a param.